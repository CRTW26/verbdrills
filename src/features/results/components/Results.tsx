import React from 'react'
import Button from 'shared/components/Button'
import { IncorrectAnswer, Score } from 'shared/types'
import { useRouter } from 'next/router'

interface Props {
  score: Score
  incorrectAnswers: IncorrectAnswer[]
  onClick: () => void
  onShowIncorrectAnswers: () => void
}

const Results: React.FC<Props> = ({
  score,
  onClick,
  onShowIncorrectAnswers,
}) => {
  const router = useRouter()

  const { total, correct, incorrect } = score

  return (
    <>
      <h1>You scored:</h1>
      <h2>{correct}</h2>
      <Button
        className="btn btn--primary"
        text="Train again"
        onClick={onClick}
      />

      <Button
        className="btn btn--primary"
        text="Go back to start"
        onClick={() => router.push(`/${router.query.language}/configure`)}
      />

      <div className="stats">
        <h3>Stats</h3>
        <h4>{`Total: ${total}`}</h4>
        <h4>{`Correct: ${correct}`}</h4>
        <h4>{`Incorrect: ${incorrect}`}</h4>
        <h4>{`Percentage: ${
          total > 0 ? Math.round((correct / total) * 100) : 0
        }%`}</h4>
        <Button
          className="btn--link"
          onClick={onShowIncorrectAnswers}
          text="Show incorrect answers"
        />
      </div>

      <style jsx>{`
        .stats {
          width: 100%;
          text-align: left;
        }

        h4 {
          margin 0.5rem 0;
        }
      `}</style>
    </>
  )
}

export default Results
