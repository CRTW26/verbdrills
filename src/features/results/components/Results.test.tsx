import { render } from '@testing-library/react'
import Results from './Results'

describe('<Results />', () => {
  test('should render', () => {
    const { asFragment } = render(
      <Results
        score={{ total: 10, correct: 5, incorrect: 5 }}
        onClick={jest.fn()}
        incorrectAnswers={[
          {
            infinitive: 'hablar',
            person: 'yo',
            answer: 'hablo',
            input: 'hable',
          },
        ]}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
