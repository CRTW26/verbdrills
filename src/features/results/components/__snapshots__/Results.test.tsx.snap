// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> should render 1`] = `
<DocumentFragment>
  <h1>
    You scored:
  </h1>
  <h2>
    5
  </h2>
  <button
    class="jsx-347471370 btn btn--primary"
  >
    Train
  </button>
  <div>
    <h3>
      Stats
    </h3>
    <h4>
      Total: 10
    </h4>
    <h4>
      Correct: 5
    </h4>
    <h4>
      Incorrect: 5
    </h4>
    <h4>
      Percentage: 50
    </h4>
  </div>
</DocumentFragment>
`;
